redwood-compose/src/commonMain/kotlin/app/cash/redwood/compose/UiConfiguration.kt

```diff
@@ -43,3 +47,64 @@ public val UiConfiguration.Companion.current: UiConfiguration
   @Composable
   @ReadOnlyComposable
   get() = LocalUiConfiguration.current
+
+/**
+ * The insets of the viewport that the composition is not safe for content because it is obscured
+ * by the system.
+ *
+ * To avoid double-consuming insets, consume them using [ConsumeInsets].
+ */
+public val LocalViewInsets: ProvidableCompositionLocal<Margin> =
+  compositionLocalWithComputedDefaultOf {
+    LocalUiConfiguration.currentValue.viewInsets
+  }
+
+/**
+ * Consume [LocalViewInsets] for the execution of [block]. This will consume all available insets
+ * unless [maximumValue] is set.
+ *
+ * The parameter to [block] is the actual insets that were consumed.
+ *
+ * Note that this function **does not** apply to the code preceding or following the caller, so it
+ * is possible for insets to be consumed multiple times. In layouts like `Box`, this may be the
+ * desired behavior.
+ */
+@Composable
+public fun ConsumeInsets(
```

Comment:
I couldn’t borrow the ComposeUI API directly 'cause it uses ComposeUI modifiers which we don’t have!